### ISSUE TITLE
We need to sync our efforts around Saltcorn evolution

### ISSUE THREAD

**Issue by AleksandrSokolov** [saltcorn/saltcorn#1223]
@glutamate @christianhugoch 

Please connect. 
I have some questions.
For example:
How to deal with noSignal?
![image](https://user-images.githubusercontent.com/327030/172070324-6caa56ad-ec7f-4a58-8fe2-b73631e1dc8a.png)

i'm suggesting to have some group to sync our efforts and cover questions about Saltcorn evolution.
For example by emails.

**Comment by AleksandrSokolov** [comment]
The problem with this new parameter noSignal that I haven't idea why it. It not given in few places where functions refersh* is used. Please give some clarification by add jsdoc description when you add new arguments to core functions and correct calls to those functiomns. Other ooption is add default value for new argument  that kept behavior of function before adding argument if it possible.

**Comment by glutamate** [comment]
Yes, this is a very good point. I will update the documentation this morning

**Comment by glutamate** [comment]
Specifically to answer your question @AleksandrSokolov, noSignal means: do not signal the refresh to other cluster processes (for multi-core - node.js is Single threaded, so to exploit multiple cores we must run multiple processes. This is coordinated by the node module cluster, using signals between the processes). In normal application code, this should be set to false (or just don't fill in the argument, which makes is falsy). This means that a signal will be sent to other cluster processes to perform the same state refresh there. Then it is important that that refresh is called with noSignal = true so we don't trigger an infinite loop of refreshes across the processes - done here: https://github.com/saltcorn/saltcorn/blob/master/packages/server/serve.js#L101.

**Comment by christianhugoch** [comment]
Hi Aleksandr, if you need to reach me, you can use the email that you see when you make git log. I just didn't want to write it directly here ...
